atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -354,9 +361,17 @@ private HikariDataSource getDataSourcePool() {
                 }
             }
         } catch (PoolInitializationException e) {
+            // Intentionally ignoring suppressed exceptions
+            // Keep duplicates to safeguard the full causality chain
+            List<Integer> vendorSqlErrorCodes = Streams.stream(
+                            Iterables.filter(Throwables.getCausalChain(e), SQLException.class))
+                    .map(SQLException::getErrorCode)
+                    .collect(Collectors.toList())
+
             log.error(
                     "Failed to initialize hikari data source",
                     SafeArg.of("connectionPoolName", connConfig.getConnectionPoolName()),
+                    SafeArg.of("vendorSqlErrorCodes", vendorSqlErrorCodes),
```

Comment:
It is possible to modify the safe argument name to allude to the fact that the list ordering matches the causal chain. However, given the target audience is the team/devs, the code will be consulted anyway. Another way of doing that is using a `LinkedList`, only if the relevant `toString` (or, whatever mechanism the logging infrastructure uses to render values) alludes to to the inherent object being linked.

It is also possible to include codes suppressed exceptions. However, I could not find a utility method in standard libraries that I am aware of to perform the operation I want. Essentially, It would amount to exploring the exception graph. Unless Oracle/Postgres use suppressed exceptions more than causes, I believe this should be a good start and we can reconsider if a use case materialized. Any thoughts @CVdV-au? 

Note than `getCausalChain` only returns a snapshot/view. If exceptions therein are mutated while it is running, results might not match the end state.